tests/rbd/rbd_rename_cloned_snapshot.py

```diff
@@ -0,0 +1,125 @@
+from random import randint
+
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_snapshot_rename_clone(rbd, pool_type, **kw):
+    """
+    This module tests snapshot rename functionality
+    Args:
+        rbd: RBD object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: test data
+    """
+    try:
+        pool = kw["config"][pool_type]["pool"]
+        image = kw["config"][pool_type]["image"]
+        snap = kw["config"][pool_type].get("snap", f"{image}_snap")
+        clone = kw["config"][pool_type].get("clone", f"{image}_clone")
+        image_spec = f"{pool}/{image}"
+        snap_rename = kw["config"][pool_type].get("snap_rename", f"{image}_new_snap")
+
+        num_of_snaps = randint(5, 20)
+        log.info(
+            f"Testing snapshot rename on {num_of_snaps} number of snaps created for image {image_spec}"
+        )
+
+        # Create snaps on original image
+        snaps_created = []
+        for i in range(1, num_of_snaps):
+            snap_name_i = f"{snap}_{i}"
+            snaps_created.append(snap_name_i)
+            if rbd.snap_create(pool, image, snap_name_i):
+                log.error(f"Creation of snapshot {snap_name_i} failed")
+                return 1
+
+        # List all snapshots created
+        listed_snaps = rbd.snap_ls(pool, image)
+        if not len(listed_snaps) == num_of_snaps - 1:
+            log.error(
+                f"Number of snapshots created {num_of_snaps} is not equal to number listed {len(listed_snaps)}"
+            )
+            return 1
+        snaps_listed = [snap["name"] for snap in listed_snaps]
+        is_not_present = [snap for snap in snaps_created if snap not in snaps_listed]
+        # snaps_created.sort() != snaps_listed.sort()
+        if is_not_present:
+            log.error(
+                f"Snaps created {snaps_created} mismatches with snaps listed {snaps_listed}"
+            )
+            return 1
+        # Add logic to verify all created snapshots are present
```

Comment:
Does it needs a `To do`?

Reply:
Removed the comment since the logic is already done. Had forgotten to remove it before.